xds/internal/balancer/clusterresolver/clusterresolver_test.go

```diff
@@ -213,15 +208,21 @@ func (s) TestSubConnStateChange(t *testing.T) {
 	defer cleanup()
 
 	builder := balancer.Get(Name)
-	edsB := builder.Build(newNoopTestClientConn(), balancer.BuildOptions{Target: resolver.Target{Endpoint: testServiceName}})
+	edsB := builder.Build(newNoopTestClientConn(), balancer.BuildOptions{Target: resolver.Target{Endpoint: testEDSServcie}})
 	if edsB == nil {
 		t.Fatalf("builder.Build(%s) failed and returned nil", Name)
 	}
 	defer edsB.Close()
 
 	if err := edsB.UpdateClientConnState(balancer.ClientConnState{
-		ResolverState:  xdsclient.SetClient(resolver.State{}, xdsC),
-		BalancerConfig: &EDSConfig{EDSServiceName: testServiceName},
+		ResolverState: xdsclient.SetClient(resolver.State{}, xdsC),
+		BalancerConfig: &LBConfig{
+			DiscoveryMechanisms: []balancerconfig.DiscoveryMechanism{{
+				Cluster:        testClusterName,
+				Type:           balancerconfig.DiscoveryMechanismTypeEDS,
+				EDSServiceName: testEDSServcie,
+			}},
+		},
```

Comment:
This identical value appears many times - factor out into a global?

Reply:
refactored to a function (with edsService string as an input)